Formula/llvm.rb

```diff
@@ -41,9 +43,15 @@ class Llvm < Formula
   on_linux do
     depends_on "pkg-config" => :build
     depends_on "binutils" # needed for gold
+    depends_on "gcc"
     depends_on "libelf" # openmp requires <gelf.h>
+    if Formula["glibc"].any_version_installed? || OS::Linux::Glibc.system_version < Formula["glibc"].version
+      depends_on "glibc"
+    end
```

Comment:
Is it worth adding DSL for this in `brew`, to simplify writing it repeatedly across different formulae?

Reply:
I think we could add one. Like the macos sdk DSLs. We can merge this PR and work on the DSL when we have time for it.

Reply:
Agreed on this, that `if` statement is quite complicated and should be simplified.  I'm not actually sure that second check is necessary - if your `glibc` version is too old, that is the first formula which gets installed when you set up Homebrew for the first time.

Reply:
> if your `glibc` version is too old, that is the first formula which gets installed when you set up Homebrew for the first time.

In this case, we should probably drop the second check. The only time that the second check is needed is when `glibc` isn't getting installed when it needed to be. However, whenever that's true, we should be fixing things in `brew` and not papering over them with workarounds in formulae.

Reply:
Yes, my thoughts exactly.  That check goes back to `llvm@7` and it may be irrelevant now.

Reply:
Without the second check this could just be
```suggestion
    depends_on "glibc" if Formula["glibc"].any_version_installed?
```
and we may not even need any new DSL for this. We would also no longer need the
```ruby
require "os/linux/glibc"
```

Reply:
I think I'm going to clean both the `require` statement and second check out of all versions of the formula in that case.

Reply:
I'm actually also wondering why the `glibc` dependency is handled in superenv whenever it's installed, rather than having to do
```ruby
depends_on "glibc" if Formula["glibc"].any_version_installed?
```
in various formulae. That can probably be addressed separately, however.

Reply:
I cleaned all of this up now (apologies again for backing up the CI queue - I should have spaced out my updates for this).  

@iMichka do you want me to clean this up in linuxbrew/core before we merge here?  Or will the changes be obvious enough based to fix when you merge from homebrew/core to linuxbrew/core.